tests/longhaul/base_object_test.go

```diff
@@ -0,0 +1,105 @@
+package longhaul
+
+import (
+	"math/rand"
+	"sync"
+	"testing"
+
+	"github.com/icrowley/fake"
+	"github.com/rook/rook/tests/framework/clients"
+	"github.com/rook/rook/tests/framework/installer"
+	"github.com/rook/rook/tests/framework/utils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"time"
+)
+
+// Set up rook cluster if necessary
+// create object store if necessary
+// create object store user if necessary
+// returns k8s helper, install helper and s3 helper for the object store that was just created.
+func setUpRook(t func() *testing.T, namespace string) (*utils.K8sHelper, *installer.InstallHelper, *clients.TestClient) {
+	kh, err := utils.CreateK8sHelper(t)
+	assert.Nil(t(), err)
+
+	i := installer.NewK8sRookhelper(kh.Clientset, t)
+	if !kh.IsRookInstalled(namespace) {
+		isRookInstalled, err := i.InstallRookOnK8sWithHostPathAndDevices(namespace, "bluestore", "/temp/rookBackup", true, 3)
+		require.NoError(t(), err)
+		require.True(t(), isRookInstalled)
+	}
+	helper, err := clients.CreateTestClient(kh, namespace)
+	if err != nil {
+		logger.Errorf("Cannot create rook test client, er -> %v", err)
+		t().FailNow()
+	}
+
+	return kh, i, helper
+}
+
+func createObjectStoreAndUser(t func() *testing.T, kh *utils.K8sHelper, tc *clients.TestClient, namespace string, storeName string, userId string, userName string) *utils.S3Helper {
+	if !isObjectStorePresent(kh, namespace, storeName) {
+		tc.GetObjectClient().ObjectCreate(namespace, storeName, 3, false, kh)
+	}
+
+	ou, err := tc.GetObjectClient().ObjectGetUser(storeName, userId)
+	if err != nil || *ou.DisplayName != userName {
+		tc.GetObjectClient().ObjectCreateUser(storeName, userId, userName)
+	}
+
+	conninfo, conninfoError := tc.GetObjectClient().ObjectGetUser(storeName, userId)
+	require.Nil(t(), conninfoError)
+	s3endpoint, _ := kh.GetRGWServiceURL(storeName, namespace)
+	s3client := utils.CreateNewS3Helper(s3endpoint, *conninfo.AccessKey, *conninfo.SecretKey)
+
+	return s3client
+
+}
+
+func isObjectStorePresent(kh *utils.K8sHelper, namespace string, storeName string) bool {
+	listOpts := metav1.ListOptions{LabelSelector: "app=rook-ceph-rgw"}
+	podList, err := kh.Clientset.Pods(namespace).List(listOpts)
+	if err == nil {
+		for _, pod := range podList.Items {
+			lables := pod.GetObjectMeta().GetLabels()
+			if lables["rook_object_store"] == storeName {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+func performObjectStoreOperations(installer *installer.InstallHelper, s3 *utils.S3Helper, bucketName string) {
+	var wg sync.WaitGroup
+	for i := 1; i <= installer.Env.LoadConcurrentRuns; i++ {
```

Comment:
what is a typical value for `LoadConcurrentRuns`? Seems like we might spend more time starting the object store and deleting it than running the operations on the object store. I would expect the longhaul tests to spend a very high percentage of the time inside the object store operations instead of a lot of times initializing and tearing down the object store

Reply:
`LoadConcurrentRuns` default value is 20, and you are right, object store initilization and teardown takes way longer than actual object store operations. I will loop through all the object store operations and sleep for a short time(1 to 2 sec).

Reply:
added another flag to the test `LoadTime`. which defaults to 120s. Each thread performs s3 or db operations for LoadTime.

Reply:
why would we sleep between the operations? During a longhaul test we trying to see if the load can be sustained for a long period of time, right? We want to know that the system can sustain enterprise-level loads. For example, why wouldn't LoadTime be something longer like an hour?

Reply:
Ok, Tweaked the params, removed sleep.